lib/builderator/tasks/clean.rb

```diff
@@ -0,0 +1,53 @@
+require 'thor'
+require_relative '../control/asg'
+require_relative '../control/ec2'
+
+module Builderator
+  module Tasks
+    class Clean < Thor
+      class_option :commit, :type => :boolean, :default => false, :desc => 'Execute cleanup'
+      class_option :filter,
+                   :type => :array,
+                   :desc => 'Key/value pairs to filter resources (--filter name foo owner_id 123456789)'
+
+      desc 'configs', 'Delete unused launch configurations'
+      def configs
+        Control::ASG.unused_launch_configs.each do |l, _|
+          say_status :remove, "Launch Configuration #{ l }", :red
+        end
+      end
+
+      desc 'images', 'Deregister unused images'
+      option 'group-by',
+             :type => :array,
+             :desc => 'Tags/properties to group images by for pruning'
```

Comment:
This is a good point. I'm thinking that we need to add a default limit to the number of resources that would be deleted. If the `unused` count exceeds that, the task will fail, requiring a human to verify the set before continuing.